Pod/Classes/PINRemoteImageManager.m

```diff
@@ -1564,6 +1540,48 @@ - (void)objectForKey:(NSString *)key options:(PINRemoteImageManagerDownloadOptio
     }
 }
 
+/// Attempt to find the task with the callbacks for the given uuid
+- (nullable PINRemoteImageTask *)_locked_taskForUUID:(NSUUID *)uuid key:(NSString * _Nullable * _Nullable)outKey
+{
+    /**
+     * Use enumerateKeysAndObjects concurrently to lookup the task.
+     * This method uses dispatch_apply under the hood and so is safe
+     * from thread explosion.
+     *
+     * Block-based enumeration should be avoided in general, but unfortunately
+     * using NSDictionary.objectEnumerator + fast enumeration results in a
+     * slowish path involving 2N calls to -nextObject, N calls to objectForKey:,
+     * and an Objective-C allocation on top of the normal fast-enumeration machinery.
+     *
+     * Thus, using -objectEnumerator is slightly slower than the fast-enumeration example
+     * at https://www.objc.io/issues/7-foundation/collections/#enumeration-and-higher-order-messaging-2
+     */
+    __block PINRemoteImageTask *result = nil;
+
+    /**
+     * A flag to protect our result.
+     * In practice there will only be one task that passes this test and so
+     * the chances of us concurrently writing to result are virtually 0 but
+     * the cost here is also virtually 0 so let's just be safe.
+     */
+    __block atomic_flag finished = ATOMIC_FLAG_INIT;
+
+    [self.tasks enumerateKeysAndObjectsWithOptions:NSEnumerationConcurrent usingBlock:^(NSString * _Nonnull key, __kindof PINRemoteImageTask * _Nonnull task, BOOL * _Nonnull stop) {
```

Comment:
Eh on second thought it won't matter much either way. String hashing is crazy fast and the imp-cache will be active so probably not worth either of our times to profile. Removing the extra enumeration and allKeys will be the lion's share